### PR TITLE
Add a vscode container dev setup

### PR DESCRIPTION
**Pull request by jheek**

### CODE REVIEW

File: dev/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// For format details, see https://aka.ms/vscode-remote/devcontainer.json or the definition README at
+// https://github.com/microsoft/vscode-dev-containers/tree/master/containers/python-3
+{
+	"name": "Python 3",
+	"context": "..",
+	"dockerFile": "Dockerfile",
+
+	// Use 'settings' to set *default* container specific settings.json values on container create.
```
Comment (marcvanzee) [inline dev/.devcontainer/devcontainer.json]: Does this work? 
https://www.quora.com/How-do-I-write-comments-inside-a-JSON-document

Comment (jheek) [reply dev/.devcontainer/devcontainer.json]: yes it is allowed in vscode. I was a bit surprised as well but having some documentation is nice

File: dev/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# How to setup a dev environment using VSCode and Docker
```
Comment (marcvanzee) [inline dev/README.md]: What I think would be nice is if you would add a line or two explaining why Docker is useful. I guess that you simulate a dev environment, so problems like different OS versions etc are avoided?

File: dev/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+1. Install Visual Studo Code (VSCode).
```
[the remark this answers is not in the document]
Comment (jheek) [reply dev/README.md]: No it is optional

File: dev/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+2. Install the `Remote - Containers` extension within VSCode.
+3. Copy `.devcontainer` to the root of your workspace folder.
+4. Make sure the source code of flax lives is checked out in `flax/`.
+5. Re-open the folder workspace using the devcontainer extension.
```
[the remark this answers is not in the document]
Comment (jheek) [reply dev/README.md]: Fair enough, the user interface is not super obvious indeed.

File: dev/README.md (hunk 2 above)
Comment (jheek) [reply dev/README.md]: Done

File: dev/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jheek) [reply dev/README.md]: done

**Comment by jheek** [comment]
@marcvanzee I resolved your issues

**Comment by jheek** [comment]
@marcvanzee I need a re-approval because the target branch changed

**Pull request merged**